relentless/potential/potential.py

```diff
@@ -1,16 +1,63 @@
 __all__ = ['PairPotential','Tabulator']
 
 import json
-
+import ast
 import numpy as np
 
 from relentless.core import PairMatrix
 from relentless.core import Variable
 
 class CoefficientMatrix(PairMatrix):
-    """ Pair coefficient matrix."""
+    """ Pair coefficient matrix.
+
+    Defines a matrix of PairMatrix objects with one or more parameters.
+    Any or all of the parameters can be initialized with a default value.
+
+    Parameters
+    ----------
+    types : array_like
+        List of types (A type must be a `str`).
+    params : array_like
+        List of parameters (A parameter must be a `str`).
+    default : `dict`
+        Initial value for any or all paramters, defaults to `None`.
+
+    Raises
+    ------
+    TypeError
+        If params does not consist of only strings
+
+    Examples
+    --------
+    Create a coefficient matrix::
+
+        m = CoefficientMatrix(types=('A','B'), params=('energy','mass'))
+
+    Create a coefficient matrix with default values::
+
+        m = CoefficientMatrix(types=('A','B'), params=('energy','mass'),
+                              default={'energy':1.0, 'mass':2.0})
+
+    Set coefficient matrix values::
+
+        m['A','A']['energy'] = 2.0
+        m['A,'B']['mass'] = 1.5
+        m['A','B']['energy'] = 3.0
```

Comment:
We should discuss how users are allowed to interact with parameters in the `CoefficientMatrix` tomorrow. The original design I had in mind was to require parameters always be given as dictionaries:
```py
m['A','B'] = {'mass': 1.5, 'energy': 3.0}
```
These dictionaries currently work in an "append" mode, which may not be ideal because it is not intuitive that parameters that are unset stay at their current value. (i.e., if you set a parameter to something, you must explicitly reset it.) A user might expect that values not set in the dict get reset to their default.

It would also be very natural for someone to try to set a key individually, as Adithya has done:
```py
m['A','B']['mass'] = 1.5
m['A','B']['energy'] = 3.0
```
We should decide if this is something we should allow. I believe that currently there is no way to prevent the user from injecting a parameter into the dict because they get access to the underlying data, and then they can do whatever they want to it.